src/util/logging.ts

```diff
@@ -0,0 +1,22 @@
+import { ChangeSet } from 'edge-sync-client'
+import { Logger } from 'pino'
+
+interface ChangeSummary {
+  [key: string]: string | null
+}
+
+/**
+ * Logs a ChangeSummary from a ChangeSet.
+ */
+export const logChangeSummary = (log: Logger, changes: ChangeSet): void => {
+  const changeSummary = Object.entries(changes).reduce<ChangeSummary>(
+    (changeSummary: ChangeSummary, [key, value]) => {
+      changeSummary[key] =
+        value?.data_base64.slice(0, Math.min(8, value?.data_base64.length)) ??
+        null
+      return changeSummary
+    },
```

Comment:
This sort of code is incredibly inefficient, like O(n^2). Every time around the loop, you end up re-copying the entire object. It's better to do this mutation-style:

```js
const out: ChangeSummary = {}
for (const key of Object.keys(changes)) {
  out[key] = changes[key]?.data_base64.slice(0, Math.min(8, changes[key]?.data_base64.length)) ??
        null
}
```

However, we can probably leave the code in place, since this is for debugging. Just notes for next time.